public/components/visualization/text2viz.tsx

```diff
@@ -0,0 +1,282 @@
+ */
+
+import {
+  EuiPageBody,
+  EuiPage,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiIcon,
+  EuiButtonIcon,
+  EuiButton,
+  EuiBreadcrumb,
+  EuiHeaderLinks,
+} from '@elastic/eui';
+import React, { useEffect, useRef, useState } from 'react';
+import { i18n } from '@osd/i18n';
+
+import { useCallback } from 'react';
+import { useObservable } from 'react-use';
+import { useMemo } from 'react';
+import { SourceSelector } from './source_selector';
+import type { DataSourceOption } from '../../../../../src/plugins/data/public';
+import chatIcon from '../../assets/chat.svg';
+import { EmbeddableRenderer } from '../../../../../src/plugins/embeddable/public';
+import {
+  useOpenSearchDashboards,
+  MountPointPortal,
+  toMountPoint,
+} from '../../../../../src/plugins/opensearch_dashboards_react/public';
+import { StartServices } from '../../types';
+import {
+  VISUALIZE_EMBEDDABLE_TYPE,
+  VisSavedObject,
+  VisualizeInput,
+} from '../../../../../src/plugins/visualizations/public';
+import './text2viz.scss';
+import { Text2VizEmpty } from './text2viz_empty';
+import { Text2VizLoading } from './text2viz_loading';
+import { Text2Vega } from './text2vega';
+import {
+  OnSaveProps,
+  SavedObjectSaveModalOrigin,
+} from '../../../../../src/plugins/saved_objects/public';
+
+export const Text2Viz = () => {
+  const [selectedSource, setSelectedSource] = useState<DataSourceOption>();
+  const {
+    services: {
+      application,
+      chrome,
+      embeddable,
+      visualizations,
+      http,
+      notifications,
+      setHeaderActionMenu,
+      overlays,
+      data,
+    },
+  } = useOpenSearchDashboards<StartServices>();
+  const [input, setInput] = useState('');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [vegaSpec, setVegaSpec] = useState<Record<string, any>>();
+  const text2vegaRef = useRef(new Text2Vega(http, data.search));
+  const status = useObservable(text2vegaRef.current.status$);
+
+  useEffect(() => {
+    const text2vega = text2vegaRef.current;
+    const subscription = text2vega.getResult$().subscribe((result) => {
+      if (result) {
+        if (result.error) {
+          notifications.toasts.addError(result.error, {
+            title: i18n.translate('dashboardAssistant.feature.text2viz.error', {
+              defaultMessage: 'Error while executing text to vega',
+            }),
+          });
+        } else {
+          setVegaSpec(result);
+        }
+      }
+    });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [http, notifications]);
+
+  const onInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+  }, []);
+
+  const onSubmit = useCallback(async () => {
+    setVegaSpec(undefined);
+    const text2vega = text2vegaRef.current;
+    if (selectedSource?.label) {
+      const dataSource = (await selectedSource.ds.getDataSet()).dataSets.find(
```

Comment:
It seems we just need data source id here, can we call `selectedSource.ds.getId()` to get the data source id?

Reply:
Actually `selectedSource.ds.getId()` is not the data source id 😢